src/filters/pixelate.ts

```diff
@@ -96,3 +123,25 @@ function getCenter(options: CenterOptions): Point {
 
   return center;
 }
+
+function getArrayOfValues(
+  image: Image,
+  channel: number,
+  options: CenterOptions,
+) {
+  let array = [];
+  for (
+    let column = options.origin.column;
+    column < options.origin.column + options.width;
+    column++
+  ) {
+    for (
+      let row = options.origin.row;
+      row < options.origin.row + options.height;
+      row++
+    ) {
+      array.push(image.getValue(column, row, channel));
+    }
+  }
+  return array;
+}
```

Comment:
you should create a new function

```
getValueFunction(algorithm) {
  switch(algorithm) {
    case 'center':
      return getCellCenter; // we return the function !!! This will be the `getValue` function
    ...
  }
}
```
This function should be called before the loop and we can then use `getValue()` inside the loop


To calculate mean you do not need to create a temporary array. It still takes time and memory.

You may also create 3 methods

- getCellCenter()
- getCellMedian()
- getCellMean()